src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -261,5 +261,52 @@ public static CSharpType FromUnion(IReadOnlyList<CSharpType> unionItemTypes, boo
 
         internal static CSharpType FromSystemType(BuildContext context, Type type, IEnumerable<ObjectTypeProperty>? backingProperties = null)
             => FromSystemType(type, context.DefaultNamespace, context.SourceInputModel, backingProperties);
+
+        public bool TryCast<T>([MaybeNullWhen(false)] out T provider) where T : TypeProvider
+        {
+            provider = null;
+            if (this.IsFrameworkType)
+                return false;
+
+            provider = this.Implementation as T;
+            return provider != null;
+        }
```

Comment:
well last time I removed this method because we could always use a is expression to get the same result:
```
if (type is {IsFrameworkType: false, Implementation: ObjectType objectType})
```
which is the same as
```
if (type.TryCast<ObjectType>(out var objectType))
```

I do not have objections on introducing this back personally

Reply:
This is not in use, introduced during the merge, I will remove it.

Reply:
removed